tests/test_app/test_app_utils.py

```diff
@@ -28,6 +50,43 @@ def test_create_temp_input(mock_upload):
         with create_temp_input(mock_upload) as tf:
             raise ValueError
     except ValueError:
-        # catch thrown thrown exception
+        # catch thrown exception
         pass
     assert not tf.is_file()
+
+
+@patch("remarx.app.utils.mo")
+def test_create_header(mock_mo):
+    """Test create_header function"""
+    # Mock the marimo functions
+    mock_mo.vstack.return_value = "mocked_header"
+    mock_mo.md.return_value = Mock()
+    mock_mo.nav_menu.return_value = Mock()
+
+    result = create_header()
+
+    mock_mo.vstack.assert_called_once()
+    assert result == "mocked_header"
+
+
+@patch("remarx.app.utils.webbrowser")
+@pytest.mark.asyncio
+async def test_lifespan(mock_webbrowser):
+    """Test lifespan context manager"""
+    from fastapi import FastAPI
+
+    # Create a mock app
+    app = FastAPI()
+
+    # Test the lifespan context manager
+    async with lifespan(app):
+        mock_webbrowser.open.assert_called_once_with("http://localhost:8000/")
```

Comment:
interesting syntax! I haven't used python async stuff yet.

Briefly made me wonder if we could call it like this in the app and pass in host and port, but I see now this is for testing.